### DISCUSSION TITLE
counting digits of a number by using for loop in python

### DISCUSSION THREAD

**Discussion by nazanintalebi** [TheAlgorithms/Python#8729] [Q&A]
HI, here are my codes :
---------------------------------------
n=int(input('please enter an integer number: '))
i=0
r=1
for i in range(0,r):
    r=n//10
    if r>=1:
        i=i+1  
print(i)
---------------------------------------
It didn't work
I can't figure it out 
please, help

**Comment by ghaianmol1122** [comment]
Here are the changes that have been made:

The for loop has been replaced with a while loop. Since the number of
iterations is not known in advance, a while loop is more appropriate in
this case.

The value of r has been initialized to n instead of 1 to ensure the loop
runs at least once.

The indentation of the code inside the loop has been fixed.

The i = i + 1 statement has been moved inside the loop, so that it gets
executed whenever r is greater than or equal to 10.

The output has been modified to display the result using print.

Now, the code will calculate and print the number of digits in the input
number correctly.


n = int(input('Please enter an integer number: '))
i = 0
r = n

while r >= 10:
    r = r // 10
    i = i + 1

print("The number of digits in", n, "is", i+1)

**Reply by nazanintalebi** [reply]
Thanks for the replay

**Comment by Feroz1522** [comment]
#empty list 
lst = []

#Input
print("enter the digit ")
Values = int(input(" >"))

#incrementing due to range function property 
Value = Values + 1

for num in range(1, Value) :
    #adding values in the list
    lst.append(num)

#printing the values on the list
print(lst)
    
#counting the digit of the numbers
print("the total counting for the digits " ,len(lst))
